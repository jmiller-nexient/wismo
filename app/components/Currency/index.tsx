import * as React from 'react';
import { FormattedNumber } from 'react-intl';

interface CurrencyProps {
    type?: string;
    value: (number | string);
}

const Currency: React.FC<CurrencyProps> = (props: CurrencyProps): React.ReactElement => {
    const {
        type = 'USD',
        value,
    } = props;

    const amount = (value && !isNaN(+value)) ? value : '0';

    return (
        <FormattedNumber
            currency={type}
            style={'currency'}
            value={+amount}
        />
    );
};

export default Currency;
