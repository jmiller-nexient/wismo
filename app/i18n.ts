/**
 * i18n.js
 *
 * This will setup the i18n language files and locale data for your app.
 *
 *   IMPORTANT: This file is used by the internal build
 *   script `extract-intl`, and must use CommonJS module syntax
 *   You CANNOT use import/export in this file.
 */
import { DEFAULT_LOCALE } from './locales';

const enTranslationMessages = require('translations/en.json');
const esTranslationMessages = require('translations/es.json');

// The purpose of this method is to apply fallback logic when a translation
// is not defined for the current locale.
// In our case, we do not want any fallback logic.
// tslint:disable-next-line: typedef
const formatTranslationMessages = (locale: string, messages) => {
  const defaultFormattedMessages =
    locale !== DEFAULT_LOCALE ? formatTranslationMessages(DEFAULT_LOCALE, enTranslationMessages) : {};

  // tslint:disable-next-line: typedef
  const flattenFormattedMessages = (formattedMessages, key) => {
    const formattedMessage =
      !messages[key] && locale !== DEFAULT_LOCALE ? defaultFormattedMessages[key] : messages[key];

    return { ...formattedMessages, [key]: formattedMessage };
  };

  return Object.keys(messages).reduce(flattenFormattedMessages, {});
};

export const translationMessages = {
  en: formatTranslationMessages('en', enTranslationMessages),
  es: formatTranslationMessages('es', esTranslationMessages),
};
