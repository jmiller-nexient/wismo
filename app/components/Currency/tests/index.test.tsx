import React from 'react';
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';

import { IntlProvider } from 'react-intl';
import { translationMessages } from 'i18n';

import Currency from '..';

describe('<Currency />', () => {
    it('should render correctly when currency has value of number 1000', () => {
        const tree: ReactTestRendererJSON | null = renderer.create(
            <IntlProvider locale="en" messages={translationMessages.en}>
                <Currency type="USD" value={1000} />
            </IntlProvider>,
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render correctly when currency has value of string ".59"', () => {
        const tree: ReactTestRendererJSON | null = renderer.create(
            <IntlProvider locale="en" messages={translationMessages.en}>
                <Currency type="USD" value=".59" />
            </IntlProvider>,
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render correctly when currency has value of string "1"', () => {
        const tree: ReactTestRendererJSON | null = renderer.create(
            <IntlProvider locale="en" messages={translationMessages.en}>
                <Currency type="USD" value="1" />
            </IntlProvider>,
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render correctly when currency has value of number -1.23', () => {
        const tree: ReactTestRendererJSON | null = renderer.create(
            <IntlProvider locale="en" messages={translationMessages.en}>
                <Currency type="USD" value={-1.23} />
            </IntlProvider>,
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
