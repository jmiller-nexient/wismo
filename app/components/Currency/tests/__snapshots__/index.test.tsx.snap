// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Currency /> should render correctly when currency has value of number -1.23 1`] = `"-$1.23"`;

exports[`<Currency /> should render correctly when currency has value of number 1000 1`] = `"$1,000.00"`;

exports[`<Currency /> should render correctly when currency has value of string ".59" 1`] = `"$0.59"`;

exports[`<Currency /> should render correctly when currency has value of string "1" 1`] = `"$1.00"`;
